app/models/concerns/findings/reiterations_oracle.rb

```diff
@@ -0,0 +1,90 @@
+module Findings::ReiterationsOracle
+  extend ActiveSupport::Concern
+
+  included do
+    scope :repeated,     -> { where     state: Finding::STATUS[:repeated] }
+    scope :not_repeated, -> { where.not state: Finding::STATUS[:repeated] }
+
+    scope :with_repeated,    -> { where.not repeated_of_id: nil }
+    scope :without_repeated, -> { where     repeated_of_id: nil }
+
+    before_save :check_for_reiteration
```

Comment:
Ya que estás clavale el:
```suggestion
    before_save :check_for_reiteration, if: :reiteration? 
```
Que quedó joya en el otro.